site/docs/plugins/i18n.md

```diff
@@ -1,105 +1,694 @@
+
+```ts
+i18n.loadLocale("en", {
+  source: `greeting = Hello { $first_name }!
+language-set = Language has been set to English!`,
+});
+```
+
+## Custom Placeables and Functions
+
+With these two options you can write your translations work. Both are provided by Fluent.
+You can read there documentation and see how they are used in the following chapter.
+
+[Placeables](https://projectfluent.org/fluent/guide/placeables.html) and [Functions](https://projectfluent.org/fluent/guide/functions.html) are two core features of the Fluent localization system.
+This plugin injects some placeables and functions into your translation context and reduces your work of managing the translation context.
+
+### Placeables
+
+Placeables are used to inject data into a translation string.
+For example, in a welcome message you may use the name of the user to greet them.
+But you can't have messages for each user with their name in your translations.
+Instead, you can pass their name in the translation context and the message will change according to the receiver. Here is how you would normally do it:
+
+`en.ftl`
+
+```fluent:no-line-numbers
+welcome = Hi there, { $name }!
+```
+
+We can pass a value for the placeable `name`, in the translation context like this:
+
+```ts:no-line-numbers
+await ctx.reply(ctx.t("welcome", { name: ctx.from.first_name }))
+```
+
+Then the message will be different for each user, with their own first names in it.
+
+#### `first_name`
+
+It is a replacement for `ctx.from.first_name`.
+So, you don't need to pass it in the context when you need it. You can just use it directly like this
+
+```fluent:no-line-numbers
+greeting = Hi { $first_name }
+```
+
+### Functions
+
+Functions are like placeables. But unlike placables, functions get executed and replaced by the value it returns in the final output.
+
+#### `CTX`
+
+This function can help you to access any properties of the bot context directly from your translations.
+
+```fluent:no-line-numbers
+id = Your user ID: { CTX("from.id") }
+```
+
+The `CTX("from.id")` will get replaced by `ctx.from.id`.
+
+Optionally, you can pass in a default value as the second argument.
+If the value is undefined, the default value is used instead.
+
+```fluent:no-line-numbers
+id = Last name: { CTX("from.last_name", "You don't have a last name.") }
+```
+
+## Listening for Translated Text
+
+We managed to send translated texts to the user.
+Now imagine, what if the user sends the bot a text?
+Yes, we use the `bot.hears` to handle them.
+But, what if the user sends a translated version of the text we're trying to listen to?
+This plugin also makes this part easy for you.
+
+```ts
+bot.hears("Hello", async (ctx) => {
+  // handle update
+});
+bot.hears("Hallo", async (ctx) => {
+  // handle update
+});
+```
+
+If those handlers are the same, we could do this:
+
+```ts
+bot.hears(["Hello", "Hallo"], async (ctx) => {
+  // handle update
+});
```

Comment:
This needs to be adjusted to `hears`. Also check if the rest of this subsection needs to be touched accordingly.

Reply:
I've tried my best on explaining this stuff short and simple as possible. Needs your review and suggestions.